src/org/scratchOrgSettingsGenerator.ts

```diff
@@ -157,42 +203,16 @@ export default class SettingsGenerator {
     }
   }
 
-  private async writeObjectSettingsIfNeeded(objectsDir: string) {
-    if (!this.objectSettingsData || !Object.keys(this.objectSettingsData).length) {
-      return;
-    }
-    for (const objectName of Object.keys(this.objectSettingsData)) {
-      const value = this.objectSettingsData[objectName];
-      // writes the object file in source format
-      const objectDir = path.join(objectsDir, upperFirst(objectName));
-      const customObjectDir = path.join(objectDir, `${upperFirst(objectName)}.object`);
-      const customObjectXml = JsonAsXml({
-        json: this.createObjectFileContent(value),
-        type: 'RecordType',
-      });
-      this.writer.addToZip(customObjectXml, customObjectDir);
-
-      if (value.defaultRecordType) {
-        const recordTypesDir = path.join(objectDir, 'recordTypes', `${upperFirst(value.defaultRecordType)}.recordType`);
-        const recordTypesFileContent = this.createRecordTypeFileContent(objectName, value);
-        const recordTypesXml = JsonAsXml({
-          json: recordTypesFileContent,
-          type: 'RecordType',
-        });
-        this.writer.addToZip(recordTypesXml, recordTypesDir);
-        // for things that required a businessProcess
-        if (recordTypesFileContent.businessProcess) {
-          const businessProcessesDir = path.join(
-            objectDir,
-            'businessProcesses',
-            `${recordTypesFileContent.businessProcess}.businessProcess`
-          );
-          const businessProcessesXml = JsonAsXml({
-            json: this.createBusinessProcessFileContent(objectName, recordTypesFileContent.businessProcess),
-            type: 'BusinessProcess',
-          });
-          this.writer.addToZip(businessProcessesXml, businessProcessesDir);
-        }
+  private async writeObjectSettingsIfNeeded(
+    objectsDir: string,
+    allRecordTypes: string[],
+    allbusinessProcesses: string[]
+  ) {
+    if (this.objectSettingsData) {
+      for (const item of Object.keys(this.objectSettingsData)) {
+        const value: ObjectSetting = this.objectSettingsData[item];
+        const fileContent = this.createObjectFileContent(upperFirst(item), value, allRecordTypes, allbusinessProcesses);
```

Comment:
examples of business logic that belong in `createObjectFileContent` are the `upperFirst`--that belongs on L254

so you can pass it the input and it provides the right output.  Otherwise, your tests require you to set upperFirst when sending it stuff

Reply:
You're confusing the function which is my fault because I've created a class method and a global function named the same way, I've renamed such functions so that not to confuse readers.